File: codegen/lift-hlp/ghidra_scripts/GetBBS.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;

import ghidra.app.cmd.disassemble.DisassembleCommand;
import ghidra.app.util.headless.HeadlessScript;
import ghidra.framework.cmd.CompoundBackgroundCommand;
import ghidra.program.model.block.BasicBlockModel;

public class GetBBS extends HeadlessScript {

	@Override
	protected void run() throws Exception {
		var args = getScriptArgs();
		
		var compoundCmd = new CompoundBackgroundCommand("Disassemble extra code", true, true);
		
		if (args.length > 1) {
			var extraCodeFilename = args[1];
			try (var fileReader = new BufferedReader(new FileReader(extraCodeFilename))) {
				String line;
				while ((line = fileReader.readLine()) != null) {
					int offset = parseInt(line);

					var cmd = new DisassembleCommand(this.toAddr(offset), currentProgram.getMemory().getAllInitializedAddressSet(), true);
					compoundCmd.add(cmd);
				}
			}
		}
		if (!compoundCmd.applyTo(currentProgram)) {
			throw new Exception("Failed to disassemble extra code");
		}
		
		
		var bbm = new BasicBlockModel(currentProgram);
		var blocks = bbm.getCodeBlocks(monitor);
		var block = blocks.next();
		
		try (var fileWriter = new FileWriter(args[0])) {
			while (block != null) {
				fileWriter.write(String.format("%d\n", block.getMinAddress().getOffset()));
				block = blocks.next();
			}
		}
	}

}

File: codegen/lift-hlp/ghidra_scripts/get_bbs.py
#TODO write a description for this script
#@author 
#@category _NEW_
#@keybinding 
#@menupath 
#@toolbar 
from ghidra.app.util.headless import HeadlessScript

class Pre(HeadlessScript):
        def run(self):
            from ghidra.program.model.block import BasicBlockModel
            from ghidra.util.task import TaskMonitor

            bbm = BasicBlockModel(currentProgram)
            blocks = bbm.getCodeBlocks(TaskMonitor.DUMMY)
            block = blocks.next()

            with open('bbs.txt', 'w') as f:
                while block:
                    f.write(str(block.minAddress.offset) + '\n')
                    block = blocks.next()

Pre().run()
